mkosi/config.py

```diff
@@ -2128,8 +2178,39 @@ def cache_manifest(self) -> dict[str, Any]:
             ),
         }
 
+    def expand_key_specifiers(self, text: str) -> str:
+        specifiers = {
+            "&": "&",
+            "d": str(self.distribution),
+            "r": self.release,
+            "a": str(self.architecture),
+            "i": self.image_id or "",
+            "v": self.image_version or "",
+            "I": self.image,
+        }
+
+        def replacer(match: re.Match[str]) -> str:
+            m = match.group("specifier")
+            if (specifier := specifiers.get(m)) is not None:
+                return specifier
+
+            logging.warning(f"Unknown specifier '&{m}' found in {text}, ignoring")
+            return ""
+
+        result = re.sub(r"&(?P<specifier>[&a-zA-Z])", replacer, text)
+
+        # Strip separator characters at the end and remove duplicate separate characters in the middle.
+        return re.sub(r"([~\-_])\1+", "\\1", result).strip("~-_")
```

Comment:
If you move `-` to either end, it doesn't need to be escaped, since it's no longer a character range
```suggestion
        return re.sub(r"([~_-])\1+", "\\1", result).strip("~-_")
```

Reply:
That being said, what's the idea behind the stripping? It works as intended when people don't mix separators, but e.g. `~-~` would not get shortened. I think keeping the separators would actually be helpful, since then even if image ID or version expand to an empty string, this is preserved when resplitting the key, e.g.:
`distribution, image_id, image_version, release = 'foo---bar'.split('-')`

Reply:
This actually isn't needed anymore, dropping